docs/source/cheat_sheet.rst

```diff
@@ -108,67 +110,71 @@ Functions
                   body=None    # type: List[str]
                   ):
        # type: (...) -> bool
-        <code>
+       <code>
 
 
 When you're puzzled or when things are complicated
 **************************************************
 
 .. code-block:: python
 
-   from typing import Union, Any, cast
+   from typing import Union, Any, List, Optional, cast
 
    # To find out what type mypy infers for an expression anywhere in
-   # your program, wrap it in reveal_type.  Mypy will print an error
+   # your program, wrap it in reveal_type().  Mypy will print an error
    # message with the type; remove it again before running the code.
-   reveal_type(1) # -> error: Revealed type is 'builtins.int'
+   reveal_type(1) # -> Revealed type is 'builtins.int'
 
-   # Use Union when something could be one of a few types.
-   x = [3, 5, "test", "fun"] # type: List[Union[int, str]]
+   # Use Union when something could be one of a few types
+   x = [3, 5, "test", "fun"]  # type: List[Union[int, str]]
 
    # Use Any if you don't know the type of something or it's too
-   # dynamic to write a type for.
-   x = mystery_function() # type: Any
+   # dynamic to write a type for
+   x = mystery_function()  # type: Any
+
+   # If you initialize a variable with an empty container or "None"
+   # you may have to help mypy a bit by providing a type annotation
+   x = []  # type: List[str]
+   x = None  # type: Optional[str]
 
-   # This is how to deal with varargs.
-   # This makes each positional arg and each keyword arg a 'str'.
+   # This makes each positional arg and each keyword arg a "str"
    def call(self, *args, **kwargs):
-            # type: (*str, **str) -> str
-            request = make_request(*args, **kwargs)
-            return self.do_api_query(request)
+       # type: (*str, **str) -> str
+       request = make_request(*args, **kwargs)
+       return self.do_api_query(request)
 
-   # Use `ignore` to suppress type-checking on a given line, when your
-   # code confuses mypy or runs into an outright bug in mypy.
-   # Good practice is to comment every `ignore` with a bug link
+   # Use a "type: ignore" comment to suppress errors on a given line,
+   # when your code confuses mypy or runs into an outright bug in mypy.
+   # Good practice is to comment every "ignore" with a bug link
```

Comment:
Up to you, but maybe we should mention the `--warn-unused-ignores` flag here (and in the Python 3 cheatsheet)?

I remember running into one or two criticisms of mypy that basically boiled down into people not knowing that this flag existed, so it might be worth surfacing it more.

(That said, I think it's also useful to keep the cheatsheets as short as possible, so that would be an argument against doing this.)

Reply:
This seems a bit too specialized for the cheat sheet.